backend/app/services/device/provider_factory.py

```
@@ -0,0 +1,103 @@
+    New device types can be added by registering their providers.
+
+    Usage:
+        # Get provider for a specific device type
+        provider = DeviceProviderFactory.get_provider(DeviceType.LOCAL)
+
+        # Register a custom provider (for future extensibility)
+        DeviceProviderFactory.register_provider(DeviceType.ALIBABA_CLOUD, AliyunDeviceProvider)
+    """
+
+    # Registry of device type -> provider class mappings
+    _providers: Dict[DeviceType, BaseDeviceProvider] = {}
```

Ruff (0.15.0)

[warning] Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)